src/analysis/asymptotics/asymptotics.lean

```diff
@@ -143,10 +143,14 @@ theorem is_o.is_O (hgf : f =o[l] g) : f =O[l] g := hgf.is_O_with.is_O
 
 lemma is_O.is_O_with : f =O[l] g → ∃ c : ℝ, is_O_with c l f g := is_O_iff_is_O_with.1
 
+theorem is_O_with.weaken' (h : is_O_with c l f g) (hg : ∀ᶠ x in l, 0 ≤ ∥g x∥) (hc : c ≤ c') :
```

Comment:
I don't understand the point of stating things assuming only `has_norm` and assumptions saying some norms are non negative.

Reply:
I left one result of this kind and renamed it to `*_aux`. It is used to prove two versions of a lemma assuming `has_norm`/`semi_normed_group` for different types.